crates/shared/src/trade_finding.rs

```diff
@@ -49,6 +120,110 @@ pub struct Trade {
+            .clearing_prices
+            .get(sell_token)
+            .context("clearing sell price missing")?
+            .to_big_rational();
+        let buy_price = self
+            .clearing_prices
+            .get(buy_token)
+            .context("clearing buy price missing")?
+            .to_big_rational();
+        let order_amount = in_amount.to_big_rational();
+
+        let out_amount = match order_kind {
+            OrderKind::Sell => order_amount
+                .mul(&sell_price)
+                .checked_div(&buy_price)
+                .context("div by zero: buy price")?,
+            OrderKind::Buy => order_amount
+                .mul(&buy_price)
+                .checked_div(&sell_price)
+                .context("div by zero: sell price")?,
+        };
+
+        big_rational_to_u256(&out_amount).context("out amount is not a valid U256")
+    }
+
+    pub fn jit_orders_executed_amounts(&self) -> Result<HashMap<H160, U256>> {
+        let mut executed_amounts: HashMap<H160, U256> = HashMap::new();
+
+        for jit_order in self.jit_orders.iter() {
+            let sell_price = self
+                .clearing_prices
+                .get(&jit_order.sell_token)
+                .context("JIT order sell token clearing price is missing")?
+                .to_big_rational();
+            let buy_price = self
+                .clearing_prices
+                .get(&jit_order.buy_token)
+                .context("JIT order buy token clearing price is missing")?
+                .to_big_rational();
+            let executed_amount = jit_order.executed_amount.to_big_rational();
+            let (executed_sell, executed_buy) = match jit_order.side {
+                Side::Sell => {
+                    let buy_amount = executed_amount
+                        .clone()
+                        .mul(&sell_price)
+                        .checked_div(&buy_price)
+                        .context("division by zero in JIT order sell")?;
+                    (executed_amount, buy_amount)
+                }
+                Side::Buy => {
+                    let sell_amount = executed_amount
+                        .clone()
+                        .mul(&buy_price)
+                        .checked_div(&sell_price)
+                        .context("division by zero in JIT order buy")?;
+                    (sell_amount, executed_amount)
+                }
+            };
+            let (executed_sell, executed_buy) = (
+                big_rational_to_u256(&executed_sell)
+                    .context("executed sell amount is not a valid U256")?,
+                big_rational_to_u256(&executed_buy)
+                    .context("executed buy amount is not a valid U256")?,
+            );
+            executed_amounts
+                .entry(jit_order.sell_token)
+                .and_modify(|executed| *executed += executed_sell)
+                .or_insert(executed_sell);
+            executed_amounts
```

Comment:
If you store the sold and the bought tokens in the same map you don't know whether tokens got bought or sold. Tokens sold (by orders) would increase the balance of the settlement while tokens bought would decrease the balance.